main/ptrarray.h

```diff
@@ -0,0 +1,50 @@
+/*
+*   Copyright (c) 1999-2002, Darren Hiebert
+*
+*   This source code is released for free distribution under the terms of the
+*   GNU General Public License version 2 or (at your option) any later version.
+*
+*   Defines external interface to resizable pointer arrays.
+*/
+#ifndef CTAGS_MAIN_PTRARRAY_H
+#define CTAGS_MAIN_PTRARRAY_H
+
+/*
+*   INCLUDE FILES
+*/
+#include "general.h"  /* must always come first */
+
+/*
+*   DATA DECLARATIONS
+*/
+
+typedef void (*ptrArrayDeleteFunc) (void *data);
+
+typedef struct sPtrArray {
+	unsigned int max;
+	unsigned int count;
+	void **array;
+	ptrArrayDeleteFunc deleteFunc;
+} ptrArray;
+
```

Comment:
I can forward-declare the struct in the header and move this to the source. Similarly with the pool.